scipy/stats/_continuous_distns.py

```diff
@@ -8846,10 +8850,16 @@ class argus_gen(rv_continuous):
 
     %(example)s
     """
+    def _logpdf(self, x, chi):
+        def _lpdf(x, chi):
+            y = 1.0 - x*x
+            A = 3*np.log(chi) - _norm_pdf_logC - np.log(_argus_phi(chi))
+            return A + np.log(x) + 0.5*np.log1p(-x*x) - chi**2 * y / 2
+
+        return _lazywhere((x > 0) & (x < 1), (x, chi), _lpdf, -np.inf)
```

Comment:
I assume this is because the support defined here:
```
argus = argus_gen(name='argus', longname="An Argus Function", a=0.0, b=1.0)
```
is inclusive of the endpoints, so you still have to treat x = 0 and x = 1?

It might be cleaner and faster to just do the calculation and suppress the warnings using `np.errstate`. I'll give it a shot and let you know what I find.

Reply:
Yeah it's faster and I think it's simpler. 
```python3
import numpy as np
from scipy.stats import argus
n = 10000
x = np.random.rand(n)
x[np.random.rand(n) > 0.9] = 1
x[np.random.rand(n) > 0.9] = 0
y = argus.logpdf(x, 0.5)
np.testing.assert_equal(y[x==0], -np.inf)
np.testing.assert_equal(y[x==1], -np.inf)
```
Current:
```
%timeit argus.logpdf(x, 0.5)
2.34 ms ± 173 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```
After suggestion:
```
%timeit argus.logpdf(x, 0.5)
881 µs ± 59.3 µs per loop (mean ± std. dev. of 7 runs, 1000 loops each)
```